src/main/scala/dxWDL/compiler/WdlCodeGen.scala

```diff
@@ -45,234 +54,226 @@ case class WdlCodeGen(verbose: Verbose,
       case WdlTypes.T_Map(keyType, valueType) =>
         val k = genDefaultValueOfType(keyType)
         val v = genDefaultValueOfType(valueType)
-        WdlValues.V_Map(Map(k -> v))
+        TAT.ExprMap(Map(k -> v), wdlType, null)
 
       // an empty array
-      case WdlTypes.T_Array(t, false) =>
-        WdlValues.V_Array(Vector.empty)
+      case WdlTypes.T_Array(_, false) =>
+        TAT.ExprArray(Vector.empty, wdlType, null)
 
       // Non empty array
       case WdlTypes.T_Array(t, true) =>
-        WdlValues.V_Array(Vector(genDefaultValueOfType(t)))
+        TAT.ExprArray(Vector(genDefaultValueOfType(t)), wdlType, null)
 
       case WdlTypes.T_Pair(lType, rType) =>
-        WdlValues.V_Pair(genDefaultValueOfType(lType), genDefaultValueOfType(rType))
+        TAT.ExprPair(genDefaultValueOfType(lType), genDefaultValueOfType(rType), wdlType, null)
```

Comment:
I can see why you are using `null` for the TextSource. It is, generally not a good idea to use `null` in scala. Of course, this use may be justified. It is up to the programmer discretion. 

In similar cases, I created a dummy token with zeros for the line and column numbers.

Reply:
yes, it was a bit lazy, but the code generator does not use the TextSource at all so in this case it is safe. I will make a note to go back and put in dummy values later.